cli/commands/genesis/root_test.go

```diff
@@ -0,0 +1,225 @@
+	statedb "github.com/berachain/beacon-kit/state-transition/core/state"
+	"github.com/berachain/beacon-kit/storage"
+	"github.com/berachain/beacon-kit/storage/beacondb"
+	"github.com/berachain/beacon-kit/storage/db"
+	depositstore "github.com/berachain/beacon-kit/storage/deposit"
+	dbm "github.com/cosmos/cosmos-db"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	pubKey1 = bytes.B48{0xff, 0xff, 0xff, 0xff}
+	creds1  = types.WithdrawalCredentials{0xaa, 0xaa, 0xaa}
+	amount1 = math.U64(2025)
+	idx1    = int(0)
+
+	pubKey2 = bytes.B48{0xee, 0xee, 0xee, 0xee}
+	creds2  = types.WithdrawalCredentials{0xbb, 0xbb, 0xbb}
+	amount2 = math.U64(5052)
+	idx2    = int(1)
+
+	emptySignature  = crypto.BLSSignature{}
+	expectedValRoot = common.Root{
+		0xa3, 0xfa, 0xd, 0x97, 0x0, 0xeb, 0xdc, 0x2c,
+		0x2, 0x1b, 0x51, 0xa1, 0xb, 0xcb, 0xb4, 0x80,
+		0x5d, 0xe6, 0x13, 0x53, 0x9a, 0x77, 0xdc, 0x65,
+		0xc7, 0x64, 0x85, 0x36, 0x6, 0xde, 0x4f, 0xa2,
+	}
+)
+
+func TestOracle(t *testing.T) {
+	cs, err := spec.MainnetChainSpec()
+	require.NoError(t, err)
+
+	deposits := genesis.Deposits{
+		{
+			Pubkey:      pubKey1,
+			Credentials: bytes.B32(creds1),
+			Amount:      amount1,
+			Signature:   emptySignature.String(), // Does not matter for validators root
+			Index:       idx1,
+		},
+		{
+			Pubkey:      pubKey2,
+			Credentials: bytes.B32(creds2),
+			Amount:      amount2,
+			Signature:   emptySignature.String(), // Does not matter for validators root
+			Index:       idx2,
+		},
+	}
+
+	cliValRoot := genesis.ValidatorsRoot(deposits, cs)
+	require.Equal(t, expectedValRoot, cliValRoot)
+}
+
+func TestStateTransitionGenesis(t *testing.T) {
+	cs, err := spec.MainnetChainSpec()
+	require.NoError(t, err)
+
+	sp, st, ds, ctx := setupState(t, cs)
+	var (
+		genDeposits = types.Deposits{
+			{
+				Pubkey:      pubKey1,
+				Credentials: creds1,
+				Amount:      amount1,
+				Signature:   emptySignature,
+				Index:       uint64(idx1),
+			},
+			{
+				Pubkey:      pubKey2,
+				Credentials: creds2,
+				Amount:      amount2,
+				Signature:   emptySignature,
+				Index:       uint64(idx2),
+			},
+		}
+		genPayloadHeader = new(types.ExecutionPayloadHeader).Empty()
+		genVersion       = version.FromUint32[common.Version](version.Deneb)
+	)
+	require.NoError(t, ds.EnqueueDeposits(ctx, genDeposits))
+	_, err = sp.InitializePreminedBeaconStateFromEth1(
+		st,
+		genDeposits,
+		genPayloadHeader,
+		genVersion,
+	)
+	require.NoError(t, err)
+
+	processorRoot, err := st.GetGenesisValidatorsRoot()
+	require.NoError(t, err)
+	require.Equal(t, expectedValRoot, processorRoot)
+}
+
+func setupState(t *testing.T, cs chain.Spec) (
+	*core.StateProcessor[*transition.Context],
+	*statedb.StateDB,
+	*depositstore.KVStore,
+	*transition.Context,
+) {
+	t.Helper()
+
+	execEngine := mocks.NewExecutionEngine(t)
+
+	mocksSigner := &cryptomocks.BLSSigner{}
+	mocksSigner.On(
```

Comment:
All of this code and the one below is currently duplicated from state processor.
I will move it to a common, testing package, to make it available to multiple packages without code duplication

Reply:
Done